### PR TITLE
Document JOIN queries when fetching models

### PR DESCRIPTION
**Pull request by ameotoko**
This adds explanation and an example for two undocumented `Model::findBy()` options. I feel like English is far from perfect in this one, so I'm open to wording suggestions😅

### CODE REVIEW

File: docs/dev/framework/models/_index.md [hunk 1]
```diff
@@ -96,6 +98,29 @@ $pages = PageModel::findBy('pid', 1, $options);
 $pages = PageModel::findByPid(1, $options);
 ```
 
+If the model has any relations defined in the [DCA](/framework/dca), setting `'eager' => true` will make Contao load all related records within the same database call, using joined query under the hood. Columns from each joined table will be prefixed with the column identifier from the main table, followed by double underscore character. This allows you to further filter your query with `having` option, like so:
```
Comment (fritzmg) [inline docs/dev/framework/models/_index.md]:
```suggestion
If the model has any relations defined in the [DCA](/framework/dca), setting `'eager' => true` will make Contao load all related records 
within the same database call by using a joined query under the hood. Columns from each joined table will be prefixed with the column 
identifier from the main table, followed by double underscore character. This allows you to further filter your query with `having` option,
for example:
```

**Review by fritzmg — CHANGES_REQUESTED** [review]

**Comment by ameotoko** [comment]
Actually, I thought that combination of

> has any relations defined

and

> only works when relation type is `hasOne` or `belongsTo`

does not make sense, so I rephrased it a little.

File: docs/dev/framework/models/_index.md [hunk 2]
```diff
@@ -96,6 +98,30 @@ $pages = PageModel::findBy('pid', 1, $options);
 $pages = PageModel::findByPid(1, $options);
 ```
 
+If the model has relations of type `hasOne` or `belongsTo` defined in the [DCA](/framework/dca), setting `'eager' => true`
+will make Contao load all related records of those types within the same database call, using joined query under the hood.
+Columns from each joined table will be prefixed with the column identifier from the main table, followed by double underscore character. This allows you to further filter your query with `having` option, for example:
```
Comment (fritzmg) [inline docs/dev/framework/models/_index.md]:
```suggestion
Columns from each joined table will be prefixed with the column identifier from the main table, followed by double underscore
character. This allows you to further filter your query with `having` option, for example:
```

**Review by fritzmg — CHANGES_REQUESTED** [review]

**Review by fritzmg — APPROVED** [review]

File: docs/dev/framework/models/_index.md [hunk 3]
```diff
@@ -96,6 +98,31 @@ $pages = PageModel::findBy('pid', 1, $options);
...
+Columns from each joined table will be prefixed with the column identifier from the main table, followed by double underscore
+character. This allows you to further filter your query with `having` option, for example:
+
+```php
+// tl_article.php
+'author' => array
+(
+    'foreignKey'              => 'tl_user.name',
+    'relation'                => array('type'=>'hasOne', 'load'=>'lazy')
+)
```
Comment (fritzmg) [inline docs/dev/framework/models/_index.md]:
```suggestion
'author' => [
    'foreignKey' => 'tl_user.name',
    'relation' => ['type'=>'hasOne', 'load'=>'lazy'],
],
```

**Review by fritzmg — CHANGES_REQUESTED** [review]

File: docs/dev/framework/models/_index.md (hunk 3 above)
Comment (ameotoko) [reply docs/dev/framework/models/_index.md]: I actually like your version more; that fragment was just copy-pasted from the core😁

Comment (fritzmg) [reply docs/dev/framework/models/_index.md]: Yes, the core still uses a legacy style.

**Review by fritzmg — APPROVED** [review]

**Comment by fritzmg** [comment]
Thank you @ameotoko :)

**Pull request merged**